email/src/main/java/org/openjdk/skara/email/Email.java

```diff
@@ -125,6 +118,48 @@ private static MboxMessage parseMboxMessage(String message) {
         return ret;
     }
 
+    /**
+     * Decode quoted printable encoding text. Non ASCII characters are encoded
+     * as series of `=XX` where `XX` is the hex value of a byte. Newlines in
+     * the encoding are escaped with `=`.
+     * @param s The string to be decoded.
+     * @param charsetName The charset name to use when converting bytes to a
+     *                    back to a String.
+     * @return A String with the decoded contents.
+     */
+    private static String decodeQuotedPrintable(String s, String charsetName) {
+        byte[] in = s.getBytes(StandardCharsets.US_ASCII);
+        // The decoded buffer can never be longer than the encoded buffer as
+        // every decoding step reduces bytes.
+        byte[] out = new byte[in.length];
+        int j = 0;
+        for (int i = 0; i < in.length; i++) {
+            if (in[i] == '=') {
+                i++;
+                switch (in[i]) {
+                    case '\n' : break;
+                    case '\r' : {
+                        if (in[i + 1] == '\n') {
+                            i++;
+                        }
+                        break;
+                    }
+                    default : {
+                        out[j++] = (byte) Integer.parseInt("" + (char) in[i++] + (char) in[i], 16);
```

Comment:
There is no boundary check here, so it always assumes there are two digits following the "=". I don't know if it's  possible for mailman server to return malformed data, but if it happens, the bot will  endlessly process the malformed input.

Reply:
Oh, I was wrong, the exception will be catched at Mbox#splitMbox(), so the bot won't process the malformed data endlessly.

Reply:
Hm, not sure what's better, ignoring the email or trying our best to handle a malformed encoding. We don't even log the issue. Should probably do that at least in Mbox#splitMbox.

Reply:
Added logging when catching exception during email parsing.